service/src/main/kotlin/app/cash/backfila/dashboard/GetBackfillRunsAction.kt

```diff
@@ -120,27 +118,42 @@ class GetBackfillRunsAction @Inject constructor(
     .list(session)
     .associateBy { it.id }
 
-  private fun partitions(
+  private data class PartitionSummary(
+    val precomputingDone: Boolean,
+    val totalComputedMatchingRecordCount: Long,
+    val totalBackfilledMatchingRecordCount: Long,
+  )
+
+  private fun partitionSummary(
     session: Session,
-    pausedBackfills: List<DbBackfillRun>
-  ) = queryFactory.newQuery<RunPartitionQuery>()
-    .backfillRunIdIn(pausedBackfills.map { it.id })
-    .list(session)
-    .groupBy { it.backfill_run_id }
+    runs: List<DbBackfillRun>,
+  ): Map<Id<DbBackfillRun>, PartitionSummary> {
+    val list = session.hibernateSession.createQuery("""
```

Comment:
wish sqldelight was set up or misk queries could support it, but whatever this works